src/pylibsshext/logging.pyx

```diff
@@ -0,0 +1,93 @@
+#
+# This file is part of the pylibssh library
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, see file LICENSE.rst in this
+# repository.
+
+import logging
+
+from pylibsshext.errors cimport LibsshSessionException
+from pylibsshext.includes cimport callbacks, libssh
+
+
+ANSIBLE_PYLIBSSH_NOLOG = logging.FATAL * 2
+ANSIBLE_PYLIBSSH_TRACE = int(logging.DEBUG / 2)
+
+
+LOG_MAP = {
+    logging.NOTSET: libssh.SSH_LOG_TRACE,
+    ANSIBLE_PYLIBSSH_TRACE: libssh.SSH_LOG_TRACE,
+    logging.DEBUG: libssh.SSH_LOG_DEBUG,
+    logging.INFO: libssh.SSH_LOG_INFO,
+    logging.WARNING: libssh.SSH_LOG_WARN,
+    logging.ERROR: libssh.SSH_LOG_WARN,
+    logging.CRITICAL: libssh.SSH_LOG_NONE,
+    ANSIBLE_PYLIBSSH_NOLOG: libssh.SSH_LOG_NONE,
+}
+
+LOG_MAP_REV = {
+    **{
+        libssh_name: py_name
+        for py_name, libssh_name in LOG_MAP.items()
+    },
+}
+
+logger = logging.getLogger("libssh")
```

Comment:
Should this be called `libssh`? or perhaps
```suggestion
_logger = logging.getLogger('ansible-pylibssh')
```
?
Should we be exposing this `_logger` object to the end-users or just have it internal?

Reply:
What if we initialize the logger object in `session.pyx` (module-level or as an instance on Session) and then pass it to `set_level()` as an argument instead of having a global thing here?

Reply:
Indeed, it should be internal. I think the problem was accessing the logger from the callback `_pylibssh_log_wrapper()`, which gets called by the libssh. I probably did not want to mess with passing it as a userdata, but it should be an option. Let me have a look into that.

Reply:
Yeah, the thing is that the logging is really not tied to the session so if we allocate the logger in session and then free it we might crash, which is why I went with the global. 
The other option would be to unset the callback + userdata when session is freed, but it would still be mess if more sessions would get allocated in one thread.

So the only viable option I see could creating some singleton Logging object, encapsulating the logging context, if you prefer.

Reply:
I think it's okay to have it global, just have it underscored so that it doesn't look like public API.